packages/host/app/components/definition-container.gts

```diff
@@ -0,0 +1,204 @@
+import Label from '@cardstack/boxel-ui/components/label';
+import Component from '@glimmer/component';
+import { on } from '@ember/modifier';
+import { action } from '@ember/object';
+import { svgJar } from '@cardstack/boxel-ui/helpers/svg-jar';
+import { Button } from '@cardstack/boxel-ui';
+import { eq, and } from '@cardstack/boxel-ui/helpers/truth-helpers';
+// boxel- specific
+import { type RealmInfo } from '@cardstack/runtime-common';
+
+export type DefinitionVariant = 'instance' | 'module';
+
+interface Signature {
+  Element: HTMLElement;
+  Args: {
+    name: string | undefined;
+    fileExtension: string;
+    realmInfo: RealmInfo | null;
+    infoText: string | undefined;
+    variant: DefinitionVariant;
+    isActive: boolean;
+    onCreate?: () => void;
+    onInherit?: () => void;
+    onDuplicate?: () => void;
+  };
+}
+
+export default class DefinitionContainer extends Component<Signature> {
+  @action
+  duplicateAction() {
+    if (this.args.onDuplicate) {
+      this.args.onDuplicate();
+    }
+  }
+
+  @action
+  createAction() {
+    if (this.args.onCreate) {
+      this.args.onCreate();
+    }
+  }
+
+  @action
+  inheritAction() {
+    if (this.args.onInherit) {
+      this.args.onInherit();
+    }
+  }
+
+  get title(): string {
+    switch (this.args.variant) {
+      case 'module':
+        return 'CARD DEFINITION';
+      case 'instance':
+        return 'CARD INSTANCE';
+      default:
+        return 'CARD DEFINITION';
+    }
+  }
+  get realmName(): string | undefined {
+    return this.args.realmInfo?.name;
+  }
+  get realmIconSrc(): string | null | undefined {
+    return this.args.realmInfo?.iconURL;
+  }
+
+  <template>
+    <div class='container' ...attributes>
+      <div class='banner {{if @isActive "active"}}'>
+        <Label class='banner-title'>
+          {{this.title}}</Label>
+        <span
+          class='banner-title'
+          data-test-definition-file-extension
+        >{{@fileExtension}}</span>
+      </div>
+      <div class='content'>
+        <div class='realm-info'>
+          <img src={{this.realmIconSrc}} alt='realm-icon' />
+          <Label
+            class='realm-name'
+            data-test-definition-realm-name
+          >{{this.realmName}}</Label>
+        </div>
+        <div class='name'>{{@name}}</div>
+      </div>
+      <div class='action-buttons'>
+        {{#if (and (eq @variant 'module') @isActive)}}
+          {{#if @onCreate}}
+            <Button class='action-button' {{on 'click' this.createAction}}>
```

Comment:
Can you simplify this to `<Button class='action-button' {{on 'click' @onCreate}}>`? Similarly for all 3 actions